lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -44,15 +44,29 @@ spec = do
         eventually $ do
             r <- request @[ApiStakePool] ctx listStakePoolsEp Default Empty
             expectResponseCode HTTP.status200 r
-            -- With the current genesis.yaml we have 1 pool with 1 lovelace, and
-            -- an epoch length of 3, meaning it will have produced either 1 or
-            -- 2 blocks in the current epoch.
+            -- With the current genesis.yaml we have 3 pools with 1 lovelace,
+            -- and an epoch length of 3.
+            --
+            -- For some reason, the first pool (the node we run), produces
+            -- blocks in 100% of the /slots/. This means it will have produced
+            -- either 1 or 2 blocks in the current epoch.
             verify r
-                [ expectListSizeEqual 1
+                [ expectListSizeEqual 3
```

Comment:
Thanks for documenting this.
It sounds abnormal to me.

Also how are the slot leaders chosen here? Is it random? If it's random or non-deterministic, then it would be difficult to assert test expectations without making a flaky test.

Reply:
> Also how are the slot leaders chosen here? Is it random?

I would have expected it to be elected in 33% of the slots with the current stake-distribution.

If we get it working, we could always adjust the stake-distribution and the expectations, even if it is random.

All the rest of the tests should of course work even if the two offline pools are elected for some slots.

Reply:
> Also how are the slot leaders chosen here? Is it random?

I have no idea, but I think the leader schedules for the entire epoch are supposed to be generated at the start of each epoch in a private sense, such that no one else can predict the slot for which your pool is elected leader.

Reply:
> Also how are the slot leaders chosen here? Is it random? If it's random or non-deterministic, then it would be difficult to assert test expectations without making a flaky test.

@rvl It is, as @Anviking said a random, unpredictable but deterministic election using multi-party computation with parameters coming from the blockchain and the previous epoch. So every node contributes to the random seed in such way that they are able to compute their own schedule but can't tell if and when others have been elected. In Ouroboros Praos, it can even happens that different pool leaders are elected for a same slot with a non negligible likelihood (which is where we get the "frequent small possible rollbacks"). 

In practice, the election result  should be somewhat proportional to the leaders' stake but if even if you are supposed to get, let's say 50% of the slots, it can happen that you aren't even elected for a single one of them in the end if you're very unlucky. But, over time (looking at the asymptotic distribution), it should converge towards the same distribution.